src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ManageContent/KeyStatisticService.cs

```diff
@@ -82,6 +82,31 @@ await _dataBlockService.IsUnattachedDataBlock(releaseId, dataBlock)
             });
     }
 
+    public async Task<Either<ActionResult, KeyStatisticTextViewModel>> CreateKeyStatisticText(
+        Guid releaseId,
+        KeyStatisticTextCreateRequest request)
+    {
+        return await _persistenceHelper.CheckEntityExists<Release>(releaseId)
+            .OnSuccess(_userService.CheckCanUpdateRelease)
+            .OnSuccess(async _ =>
+            {
+                var keyStatisticText = _mapper.Map<KeyStatisticText>(request);
+                keyStatisticText.ReleaseId = releaseId;
+                keyStatisticText.CreatedById = _userService.GetUserId();
+
+                var order = await _context.KeyStatistics
+                    .Where(ks => ks.ReleaseId == releaseId)
+                    .Select(ks => ks.Order)
+                    .MaxAsync(order => (int?)order);
+                keyStatisticText.Order = order.HasValue ? order.Value + 1 : 0;
```

Comment:
I'd be tempted to write:

```
keyStatisticText.Order = (await _context
    .KeyStatistics
    .Where(ks => ks.ReleaseId == releaseId)
    .Select(ks => ks.Order)
    .MaxAsync(order => (int?)order)) ?? 0;
```

for brevity.

Reply:
(not much difference to be fair!)

Reply:
I don't think this would be right, as we still need to add the `+ 1` on to the existing max value, if it has one.

Reply:
No worries - I didn't consider the `+1`